Source/WebCore/platform/audio/cocoa/PlatformRawAudioDataCocoa.cpp

```diff
@@ -0,0 +1,320 @@
+    case AudioSampleFormat::S32:
+        return AudioStreamDescription::Int32;
+    case AudioSampleFormat::F32:
+        return AudioStreamDescription::Float32;
+    case AudioSampleFormat::U8Planar:
+        return AudioStreamDescription::Uint8;
+    case AudioSampleFormat::S16Planar:
+        return AudioStreamDescription::Int16;
+    case AudioSampleFormat::S32Planar:
+        return AudioStreamDescription::Int32;
+    case AudioSampleFormat::F32Planar:
+        return AudioStreamDescription::Float32;
+    }
+}
+
+static CAAudioStreamDescription::IsInterleaved interleavedFormat(AudioSampleFormat format)
+{
+    return isAudioSampleFormatInterleaved(format) ? CAAudioStreamDescription::IsInterleaved::Yes : CAAudioStreamDescription::IsInterleaved::No;
+}
+
+static RetainPtr<CMSampleBufferRef> createSampleBuffer(const CAAudioStreamDescription& description, const CMTime& time, size_t numberOfFrames, WebAudioBufferList* list)
+{
+    CMAudioFormatDescriptionRef rawFormatDescription;
+    if (PAL::CMAudioFormatDescriptionCreate(kCFAllocatorDefault, &description.streamDescription(), 0, nullptr, 0, nullptr, nullptr, &rawFormatDescription))
+        return nullptr;
+    RetainPtr formatDescription = adoptCF(rawFormatDescription);
+
+    CMSampleBufferRef rawSampleBuffer;
+    if (PAL::CMAudioSampleBufferCreateWithPacketDescriptions(kCFAllocatorDefault, nullptr, false, nullptr, nullptr, rawFormatDescription, numberOfFrames, time, nullptr, &rawSampleBuffer))
+        return nullptr;
+    auto sampleBuffer = adoptCF(rawSampleBuffer);
+
+    if (list) {
+        if (PAL::CMSampleBufferSetDataBufferFromAudioBufferList(sampleBuffer.get(), kCFAllocatorDefault, kCFAllocatorDefault, kCMSampleBufferFlag_AudioBufferList_Assure16ByteAlignment, list->list()))
+            return nullptr;
+        return sampleBuffer;
+    }
+
+    auto result = WebAudioBufferList::createWebAudioBufferListWithBlockBuffer(description, numberOfFrames);
+    if (!result)
+        return nullptr;
+
+    auto [newList, blockBuffer] = WTFMove(*result);
+    if (PAL::CMSampleBufferSetDataBuffer(rawSampleBuffer, blockBuffer.get()))
+        return nullptr;
+    return sampleBuffer;
+}
+
+RefPtr<PlatformRawAudioData> PlatformRawAudioData::create(std::span<const uint8_t> sourceData, AudioSampleFormat format, float sampleRate, int64_t timestamp, size_t numberOfFrames, size_t numberOfChannels)
+{
+    CAAudioStreamDescription inputDescription(sampleRate, numberOfChannels, audioSampleFormatToPCMFormat(format), interleavedFormat(format));
+    if (!WebAudioBufferList::isSupportedDescription(inputDescription, numberOfFrames)) {
+        RELEASE_LOG_ERROR(MediaStream, "PlatformRawAudioData::create not supported configuration");
+        return nullptr;
+    }
+
+    WebAudioBufferList inputList = inputDescription;
+    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+    uint8_t* data = const_cast<uint8_t*>(sourceData.data());
+    size_t sizePlane = numberOfFrames * inputDescription.bytesPerFrame();
```

Comment:
Is there a risk of overflow?

Reply:
> Is there a risk of overflow?

no. The checks are performed in `computeCopyElementCount` and we throw if any overflow in the calculation occur before being called here. And the fact that we could create the sourceData in the first place (which is even wider) indicates that there's no overflow.

I could add an assert.

Or maybe do add a check just to make it explicitly clear to the reader that it can't happen